### PR TITLE
Adding ability to override number of radial bases

### PR DESCRIPTION
**Pull request by rosecers**
Quick implementation of what @arthur-lin1027 and I discussed. Still keeps default behavior intact, but adds a manual override if wanted.

### CODE REVIEW

**Comment by arthur-lin1027** [comment]
I think this looks good. However, I'm unsure if I necessarily like the feature of inputting a custom list of max_radial

**Comment by arthur-lin1027** [comment]
Additionally, when using a fixed number of bases, it unfortunately still does not learn lennard jones successfully. I think this motivates two directions
1) Ensure my orthonormalization code is actually correct (maybe write some more tests??)
2) coupling the radial gaussian width with n and/or l

**Comment by rosecers** [comment]
@kvhuguenin thoughts?

File: anisoap/representations/radial_basis.py [hunk 1]
```diff
@@ -207,6 +229,16 @@ def orthonormalize_basis(self, features: TensorMap):
             l_2n_arr = l + 2 * n_arr
             # normalize all the GTOs by the appropriate prefactor first, since the overlap matrix is in terms of
             # normalized GTOs
+            # Ensure that each gto width is the cutoff_radius * sqrt(n) / nmax. If n < 1, then take n = 1.
```
Comment (arthur-lin1027) [inline anisoap/representations/radial_basis.py]: Also going to revert these changes too, so this PR strictly shows the manual radial.

**Review by arthur-lin1027 — APPROVED** [review]
I think we're good. @rosecers I messed with this a lot so let's make sure this PR is pretty clean.

File: anisoap/representations/radial_basis.py [hunk 2]
```diff
@@ -136,7 +154,9 @@ def get_num_radial_functions(self):
     # by a precision matrix (inverse of covariance) and its center.
     # The current function computes the covariance matrix and the center
     # for the provided parameters as well as choice of radial basis.
-    def compute_gaussian_parameters(self, r_ij, lengths, rotation_matrix):
+    def compute_gaussian_parameters(
+        self, r_ij, lengths, rotation_matrix, radial_gaussian_width=None
+    ):
```
Comment (rosecers) [inline anisoap/representations/radial_basis.py:159]: Is this meant to be an edit for later?

Comment (arthur-lin1027) [reply anisoap/representations/radial_basis.py:159]: yes it leaked through. Let's retain the original version, and introduce these changes in the "couple sigma to n and l" commit

Comment (arthur-lin1027) [reply anisoap/representations/radial_basis.py]: Restored.

**Review by arthur-lin1027 — APPROVED** [review]
These changes were necessary when incorporating constant n!

File: anisoap/representations/ellipsoidal_density_projection.py [hunk 3]
```diff
@@ -120,7 +120,7 @@ def pairwise_ellip_expansion(
                     )
 
                     moments = compute_moments_inefficient_implementation(
-                        precision, center, maxdeg=lmax + np.max(num_ns)
+                        precision, center, maxdeg=maxdeg
```
Comment (rosecers) [inline anisoap/representations/ellipsoidal_density_projection.py:123]: looks like another remnant of future PRs

Comment (arthur-lin1027) [reply anisoap/representations/ellipsoidal_density_projection.py:123]: this is not a remnant, it's necessary. Having line 78 and line 123 will ensure together makes it so that we always calculate the right number of moments.

Comment (rosecers) [reply anisoap/representations/ellipsoidal_density_projection.py:123]: :+1:
